awx_collection/plugins/inventory/tower.py

```diff
@@ -138,13 +148,20 @@ def parse(self, inventory, loader, path, cache=True):
         # Read inventory from tower server.
         # Note the environment variables will be handled automatically by InventoryManager.
         tower_host = self.get_option('host')
+        token_auth = self.get_option('token_auth')
         if not re.match('(?:http|https)://', tower_host):
             tower_host = 'https://{tower_host}'.format(tower_host=tower_host)
 
-        request_handler = Request(url_username=self.get_option('username'),
-                                  url_password=self.get_option('password'),
-                                  force_basic_auth=True,
-                                  validate_certs=self.get_option('validate_certs'))
+        if not token_auth:
+            request_handler = Request(url_username=self.get_option('username'),
+                                      url_password=self.get_option('password'),
+                                      force_basic_auth=True,
+                                      validate_certs=self.get_option('validate_certs'))
+        else:
+            token = self.get_option('password')
+            token_header = 'Bearer ' + token
```

Comment:
I know it might sound trivial, but we really want to consolidate to the same code path as the modules which does many of the same things:

https://github.com/ansible/awx/blob/2855be9d260e6c597031e018ae3ae4fb5725425b/awx_collection/plugins/module_utils/tower_api.py#L316